sklearn/linear_model/tests/test_logistic.py

```diff
@@ -2537,3 +2547,70 @@ def test_liblinear_multiclass_warning(Estimator):
     )
     with pytest.warns(FutureWarning, match=msg):
         Estimator(solver="liblinear").fit(iris.data, iris.target)
+
+
+@pytest.mark.parametrize("binary", [False, True])
+@pytest.mark.parametrize("use_str_y", [False, True])
+@pytest.mark.parametrize(
+    "array_namespace, device_, dtype_name",
+    yield_namespace_device_dtype_combinations(),
+    ids=_get_namespace_device_dtype_ids,
+)
+def test_logistic_regression_array_api_compliance(
+    binary, use_str_y, array_namespace, device_, dtype_name
+):
+    xp = _array_api_for_tests(array_namespace, device_)
+    n_samples, _ = iris.data.shape
+    X_np = iris.data.astype(dtype_name)
+    X_xp = xp.asarray(X_np, device=device_)
+    if use_str_y:
+        if binary:
+            target = (iris.target > 0).astype(np.int64)
+            target = np.array(["setosa", "not-setosa"])[target]
+        else:
+            target = iris.target_names[iris.target]
+        y_np = target.copy()
+        y_xp_or_np = np.asarray(y_np, copy=True)
+    else:
+        if binary:
+            target = (iris.target > 0).astype(np.int64)
+        else:
+            target = iris.target
+        y_np = target.astype(dtype_name)
+        y_xp_or_np = xp.asarray(y_np, device=device_)
+
+    lr_np = LogisticRegression(C=n_samples, solver="lbfgs", max_iter=200).fit(
```

Comment:
I think we should try with stronger regularization (e.g. the default C=1) to see if the problem is better conditioned and allow float32-only device computation to converge to the same value as numpy on CPU.

Reply:
If that works, we could parametrize this test to run for different values of C and xfail the float32 device for the highest C values.